draft-ietf-quic-http.md

```diff
@@ -99,49 +99,45 @@ defined in {{!RFC2119}}.
 
 # QUIC Advertisement
 
-A server advertises that it can speak HTTP/QUIC via the Alt-Svc ({{!RFC7838}})
-HTTP response header (or the semantically equivalent Alt-Svc HTTP/2 Extension
-Frame Type), using the ALPN token defined in {{connection-establishment}}.
+An HTTP origin advertises the availability of an equivalent HTTP/QUIC endpoint
+via the Alt-Svc HTTP response header or the HTTP/2 ALTSVC frame ({{!RFC7838}}),
+using the ALPN token defined in {{connection-establishment}}.
 
-Thus, a server could indicate in an HTTP/1.1 or HTTP/2 response that HTTP/QUIC
-was available on UDP port 443 by including the following header in any
-response:
+For example, an origin could indicate in an HTTP/1.1 or HTTP/2 response that
+HTTP/QUIC was available on UDP port 443 at the same hostname by including the
+following header in any response:
 
     Alt-Svc: hq=":443"
 
+On receipt of an Alt-Svc header indicating HTTP/QUIC support, a client MAY
+attempt to establish a QUIC connection to the indicated host and port and, if
+successful, send HTTP requests using the mapping described in this document.
+
+Connectivity problems (e.g. firewall blocking UDP) can result in QUIC connection
+establishment failure, in which case the client SHOULD continue using the
+existing connection or try another alternative endpoint offered by the origin.
+
+
 ## QUIC Version Hints {#alt-svc-version-hint}
 
-This document defines the "v" parameter for Alt-Svc, which is used to provide
-version-negotiation hints to HTTP/QUIC clients. Syntax:
+This document defines the "quic" parameter for Alt-Svc, which MAY be used to
+provide version-negotiation hints to HTTP/QUIC clients. QUIC versions are
+four-octet sequences with no additional constraints on format. Syntax:
 
-    v = version
-    version = DQUOTE ( "c" version-string / "x" version-number ) DQUOTE
-    version-string = 4tchar; token-safe QUIC version
+    quic = version-number
     version-number = 1*8HEXDIG; hex-encoded QUIC version
 
-When multiple versions are supported, the "v" parameter MAY be repeated multiple
-times in a single Alt-Svc entry.  For example, if a server supported both
-version "Q034" and version 0x00000001, it would specify the following header:
+When multiple versions are supported, the "quic" parameter MAY be repeated
+multiple times in a single Alt-Svc entry.  For example, if a server supported
+both version "Q034" and version 0x00000001, it could specify the following
```

Comment:
But then we'd still have to give them something to replace it with, and keeping the "how to do multiple" is worthwhile.  I've expanded slightly, to refer to it as the version that would be rendered in ASCII as `Q034`.  Given that we allow pretty free-form experimentation in that region, I don't know that we need to elucidate *why* a server might support an experimental version.